code/game/machinery/turrets.dm

```diff
@@ -789,6 +789,17 @@
 		target = pick(pos_targets)
 	return target
 
+/obj/structure/turret/gun_turret/proc/choose_roulette_projectile()
+	var/chosen_projectile = pick(available_projectiles)
+	for(var/I in restricted_roulette_projectiles)
+		if(chosen_projectile == I)
+			choose_roulette_projectile()
```

Comment:
The smart move would be to generate a pre-restricted list of available_projectiles down in the New() proc, where available_projectiles is already being generated. One time cost (when the gun/turret is made) vs this proc (which would be called in the middle of ingame action), and the added bonus of saving ram from the smaller list.

Reply:
Yeah I figured this would be the better approach but just wanted to use the existing list (and I am lazy)! I'll update it